python/ray/tune/commands.py

```diff
@@ -154,6 +158,11 @@ def list_trials(experiment_path,
         checkpoints_df["logdir"] = checkpoints_df["logdir"].str.replace(
             experiment_path, '')
 
+    if filter:
+        filter_str = "checkpoints_df[checkpoints_df['{}'] {} {}]".format(
+            *filter)
+        checkpoints_df = eval(filter_str)
```

Comment:
I don't think using `eval` is a good idea here. Generally speaking, using eval can be insecure and hard to maintain, and in almost every case there's a way to achieve the same functionality without eval. In this case, you can use some kind of simple filter language and filter the dataframe based on that. The user could provide a filter operation of the form `"<column> <operation> <value>"` (as a string), then we would convert that (without `eval`) into some kind of filter representation and then use it to filter the dataframe.

This should be easy to implement if you start by allowing just single filter, and later on you can expand it to work with more complicated filters (for example allowing ANDs/ORs, nesting things, etc.)

Reply:
Here's a simple example:
```python
import operator

import numpy as np
import pandas as pd


OPERATORS = {
    '<': operator.lt,
    '<=': operator.le,
    '==': operator.eq,
    '!=': operator.ne,
    '>=': operator.ge,
    '>': operator.gt,
}


def main():
    dataframe = pd.DataFrame({'a': np.arange(10), 'b': np.arange(10, 20)})
    filter_string = "a <= 5"

    column_name, operator_str, value_str = filter_string.split(" ")
    operator = OPERATORS[operator_str]
    value = float(value_str)
    filtered_index = operator(dataframe[column_name], value)
    filtered_dataframe = dataframe[filtered_index]
    print(filtered_dataframe)


if __name__ == '__main__':
    main()
```